ts/RootContainer.tsx

```diff
@@ -118,6 +124,15 @@ class RootContainer extends React.PureComponent<Props> {
     // FIXME: perhaps instead of navigating to a "background"
     //        screen, we can make this screen blue based on
     //        the redux state (i.e. background)
+    const isAppOutOfDate = !isVersionAppSupported(
```

Comment:
If you want to try a more functional approach you could get this value as follows

```typescript
const isAppOutOfDate = fromNullable(this.props.backendInfo)
      .map(bi => {
        const minAppVersion = Platform.select({
          ios: bi.min_app_version.ios,
          android: bi.min_app_version.android
        });
        return !isVersionAppSupported(minAppVersion, DeviceInfo.getVersion());
      })
      .getOrElse(false);
```
in this way you can avoid the first parameter of **isVersionAppSupported** to be `undefined`
Indeed this an application logic not of that function. The responsibility of **isVersionAppSupported** should be: say if the app is supported by providing an upper bound version